llvm/test/CodeGen/AIE/aie2/GlobalISel/dead-mi-elim.mir

```diff
@@ -547,3 +536,106 @@ body:             |
+  ; CHECK-NEXT:   [[COPY:%[0-9]+]]:er = COPY $r1
+  ; CHECK-NEXT:   $crsat = COPY [[MOV_RLC_imm10_pseudo]]
+  ; CHECK-NEXT:   $crrnd = COPY [[COPY]]
+  ; CHECK-NEXT:   PseudoJ_jump_imm %bb.1
+  ; CHECK-NEXT: {{  $}}
+  ; CHECK-NEXT: bb.1:
+  ; CHECK-NEXT:   PseudoRET implicit $lr
+  bb.0.entry:
+    successors: %bb.1(0x80000000); %bb.1(100.00%)
+    liveins: $r1, $p0
+    %0:er = MOV_RLC_imm10_pseudo 0
+    %1:er = COPY $r1
+    $crsat = COPY %0:er
+    $crrnd = COPY %1:er
+    PseudoJ_jump_imm %bb.1
+  bb.1:
+    %5:er = MOV_RLC_imm10_pseudo 23
+    PseudoRET implicit $lr
+...
+
+# Test 8 - Test for liveness of control registers across basic blocks with use in bb.1.
+---
+name:            live_control_regs
+alignment:       16
+legalized:       true
+regBankSelected: true
+selected:        true
+tracksRegLiveness: true
+body:             |
+  ; CHECK-LABEL: name: live_control_regs
+  ; CHECK: bb.0.entry:
+  ; CHECK-NEXT:   successors: %bb.1(0x80000000)
+  ; CHECK-NEXT:   liveins: $r1, $p0
+  ; CHECK-NEXT: {{  $}}
+  ; CHECK-NEXT:   [[MOV_RLC_imm10_pseudo:%[0-9]+]]:er = MOV_RLC_imm10_pseudo 0
+  ; CHECK-NEXT:   [[COPY:%[0-9]+]]:er = COPY $r1
+  ; CHECK-NEXT:   $crsat = COPY [[MOV_RLC_imm10_pseudo]]
+  ; CHECK-NEXT:   $crupssign = COPY [[COPY]]
+  ; CHECK-NEXT:   PseudoJ_jump_imm %bb.1
+  ; CHECK-NEXT: {{  $}}
+  ; CHECK-NEXT: bb.1:
+  ; CHECK-NEXT:   liveins: $p0
+  ; CHECK-NEXT: {{  $}}
+  ; CHECK-NEXT:   [[COPY1:%[0-9]+]]:mss = COPY [[MOV_RLC_imm10_pseudo]]
+  ; CHECK-NEXT:   [[COPY2:%[0-9]+]]:ep = COPY $p0
+  ; CHECK-NEXT:   [[VLDA_UPS_S32_D8_ag_idx_imm:%[0-9]+]]:acc1024 = VLDA_UPS_S32_D8_ag_idx_imm [[COPY1]], [[COPY2]], 0, implicit-def $srups_of, implicit $crsat, implicit $crupssign
+  ; CHECK-NEXT:   $crupssign = MOV_scalar_imm10_pseudo 0
+  bb.0.entry:
+    successors: %bb.1(0x80000000); %bb.1(100.00%)
+    liveins: $r1, $p0
+    %0:er = MOV_RLC_imm10_pseudo 0
+    %1:er = COPY $r1
+    $crsat = COPY %0:er
+    $crupssign = COPY %1:er
+    PseudoJ_jump_imm %bb.1
+
+  bb.1:
+    liveins: $p0
+    %2:mss = COPY %0:er
+    %3:ep = COPY $p0
+    %4:acc1024 = VLDA_UPS_S32_D8_ag_idx_imm %2:mss, %3:ep, 0, implicit-def $srups_of, implicit $crsat, implicit $crupssign
+    $crupssign = MOV_scalar_imm10_pseudo 0
+...
+
+# Test 9 - Test for liveness of control registers across call boundaries.
+---
+name:            live_control_regs_call
+alignment:       16
+legalized:       true
+regBankSelected: true
+selected:        true
+tracksRegLiveness: true
+body:             |
+  bb.0.entry:
+    liveins: $r1, $p0
+    ; CHECK-LABEL: name: live_control_regs_call
+    ; CHECK: liveins: $r1, $p0
+    ; CHECK-NEXT: {{  $}}
+    ; CHECK-NEXT: [[MOV_RLC_imm10_pseudo:%[0-9]+]]:er = MOV_RLC_imm10_pseudo 0
+    ; CHECK-NEXT: $crsat = COPY [[MOV_RLC_imm10_pseudo]]
+    ; CHECK-NEXT: $crrnd = COPY [[MOV_RLC_imm10_pseudo]]
+    ; CHECK-NEXT: PseudoJL 0
+    %0:er = MOV_RLC_imm10_pseudo 0
+    $crsat = COPY %0:er
+    $crrnd = COPY %0:er
+    PseudoJL 0
```

Comment:
That test seems to already work even without the fix?

Reply:
Maybe you need to add a regmask?

Reply:
right, added it.